File: Scripts/thread.cpp
/**
 * @Author: @IamRezaMousavi
 * @Date:   2022-03-08 23:45:15
 * @Last Modified by:   @IamRezaMousavi
 * @Last Modified time: 2022-03-08 23:49:56
 */

#include <iostream>
#include <thread>

using namespace std;

thread::id mainID = this_thread::get_id();

void printThreadId() {
  cout << this_thread::get_id() << endl;
}

int main(int argc, const char *argv[]) {
  cout << mainID << endl;

  thread t(printThreadId);

  cout << this_thread::get_id() << endl;

  t.join();
  return 0;
}

File: python-programs/Programs/thread.py
# @Author: @IamRezaMousavi
# @Date:   2022-03-20 22:45:20
# @Last Modified by:   @IamRezaMousavi
# @Last Modified time: 2022-03-20 22:51:34

import threading


def bigThread(threadName):
    for i in range(5):
        print(f'\tThread {threadName} -> {i} is running...')


threads = []
for i in range(4):
    t = threading.Thread(target=bigThread, args=(str(i)))
    t.start()
    threads.append(t)

for thread in threads:
    t.join()
